### PR TITLE
[SYCL][CUDA] libdevice: add builtins

### PR DESCRIPTION
**Pull request by hiaselhans**
Add some scalar math builtins to ptx/libdevice:

 - acos
 - asin
 - atan
 - cbrt
 - ceil
 - cos
 - erf
 - fabs
 - floor
 - fmax
 - fmin
 - lgamma
 - sqrt

Signed-off-by: hiaselhans <simon.klemenc@gmail.com>

### CODE REVIEW

**Comment by bader** [comment]
+@Naghasan

File: libclc/ptx-nvidiacl/libspirv/math/cos.cl [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#define __CLC_FUNCTION __spirv_ocl_cos
+#define __CLC_BUILTIN __nv_cos
+#include "unary_builtin.inc"
+
+#define __CLC_FUNCTION __spirv_ocl_cosh
+#define __CLC_BUILTIN __nv_cosh
+#include "unary_builtin.inc"
+
+/*
+Linking globals named '_Z17__spirv_ocl_cospif': symbol multiply defined!
```
Comment (Naghasan) [inline libclc/ptx-nvidiacl/libspirv/math/cos.cl:22]: can you remove the dead code please ?

File: libclc/ptx-nvidiacl/libspirv/math/atan.cl [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#include "../../../generic/lib/clcmacro.h"
+#include "../../include/libdevice.h"
+#include <spirv/spirv.h>
+
+#define __CLC_FUNCTION __spirv_ocl_atan
+#define __CLC_BUILTIN __nv_atan
+#include "unary_builtin.inc"
+
+/*
+TBD
```
Comment (Naghasan) [inline libclc/ptx-nvidiacl/libspirv/math/atan.cl:18]: same, can you remove the dead code please

File: libclc/ptx-nvidiacl/lib/SOURCES [hunk 3]
```diff
@@ -1 +1 @@
-mem_fence/fence.cl
+mem_fence/fence.cl
```
Comment (Naghasan) [inline libclc/ptx-nvidiacl/lib/SOURCES:1]: can you revert your change, that's unnecessary

Comment (hiaselhans) [reply libclc/ptx-nvidiacl/lib/SOURCES]: i don't know why it happened. undone... thx

File: libclc/ptx-nvidiacl/libspirv/math/cos.cl (hunk 1 above)
Comment (hiaselhans) [reply libclc/ptx-nvidiacl/libspirv/math/cos.cl]: Done.
For the future, do you think we should keep track of what we have and what's left to do in some file?
like in [cuda-vs-opencl-math-builtin-precisions](sycl/doc/cuda/cuda-vs-opencl-math-builtin-precisions.md)

File: libclc/ptx-nvidiacl/libspirv/math/atan.cl (hunk 2 above)
Comment (hiaselhans) [reply libclc/ptx-nvidiacl/libspirv/math/atan.cl]: done

**Comment by hiaselhans** [comment]
moving the comment so the pr message can be used:

This PR adds a few of nvcc's builtins to ptx//libdevice:

sycl/test/built-ins/scalar_math.cpp test is split in two parts:

    scalar_math.cpp -> supported by cuda (no xfail)
    scalar_math_2.cpp -> unsupported

## Missing float builtins:

    atan2
    copysign
    fdim
    fma

for these it fails with
Unresolved extern function '_Z15__spirv_ocl_fmafff'

## Integer Builtins

-> need to make something similar too binary_builtin.inc for uint, int, long, ulong

## Geometry Functions

-> these aren't within _nv*. Maybe there is a way to reuse generic functions for these?

File: libclc/ptx-nvidiacl/libspirv/math/cos.cl (hunk 1 above)
Comment (Naghasan) [reply libclc/ptx-nvidiacl/libspirv/math/cos.cl]: I have them locally, I'm finishing the patch formatting (some are already opened). There is also binding test that were added.

**Pull request merged**